test/archetype.test.ts

```diff
@@ -62,6 +63,7 @@ describe("Factory", function () {
         //   mintDiscount: number;
         // }];
       },
+      blacklisted: ethers.constants.HashZero,
```

Comment:
What happens when you omit this key from the config object? 

Ideally it shouldn't matter and should simply be preinitialized to the default value.

Reply:
Tests revert if a value isn't provided.

I noticed that even though these values are commented as optional in `Config` they're also provided everywhere in the Config in the tests.

<img width="394" alt="image" src="https://github.com/guminc/contracts/assets/14217867/d5842947-0cc1-49ba-a829-cfa008de769b">

Reply:
ok thanks. we'll have to confirm the impact it will have on frontend. Changing the Config interface might impact compatibility with existing Contract Factory. Worst case scenario is to redeploy the factory, but just need to double-check on testnet first. 

@kneelsdev please review as well.